worker/lease/manager.go

```diff
@@ -627,7 +627,9 @@ func (manager *Manager) pinned(namespace, modelUUID string) map[string][]string
 func (manager *Manager) leases(namespace, modelUUID string) map[string]string {
 	leases := make(map[string]string)
 	for key, lease := range manager.config.Store.Leases() {
-		leases[key.Lease] = lease.Holder
+		if key.Namespace == namespace && key.ModelUUID == modelUUID {
+			leases[key.Lease] = lease.Holder
+		}
 	}
 	return leases
```

Comment:
Do you want to return an `ErrNotFound` if it's empty?

Reply:
No, ErrNotFound would suggest that something's gone wrong (you've passed a bad ID or something), but it would be entirely normal for the leadership leases map to be empty for a model before you've deployed anything.

Reply:
An empty map is sufficient.